lang/compiler.go

```diff
@@ -391,6 +391,8 @@ func (p *AstProgram) getSpec(ctx context.Context, runtime flux.Runtime, alloc *m
 }
 
 func (p *AstProgram) Start(ctx context.Context, alloc *memory.Allocator) (flux.Query, error) {
+	deps := execdeps.NewExecutionDependencies(alloc, &p.Now, p.Logger)
+	ctx = deps.Inject(ctx)
```

Comment:
Can you move these lines down to line 411/413? We want to inject the dependencies in the context after the call to `getSpec` so that the correct now time is used.